addons/mail/models/mail_notification.py

```diff
@@ -41,14 +45,19 @@ class MailNotification(models.Model):
         ("mail_email_invalid", "Invalid email address"),
         ("mail_email_missing", "Missing email address"),
         ("mail_smtp", "Connection failed (outgoing mail server problem)"),
-        ], string='Failure type')
+        # mass mode
+        ("mail_bl", "Blacklisted Address"),
+        ("mail_optout", "Opted Out"),
+        ("mail_dup", "Duplicated Email")], string='Failure type')
     failure_reason = fields.Text('Failure reason', copy=False)
 
     _sql_constraints = [
-        # email notification: partner is required
-        ('notification_partner_required',
-         "CHECK(notification_type NOT IN ('email', 'inbox') OR res_partner_id IS NOT NULL)",
-         'Customer is required for inbox / email notification'),
+        ('notification_identifier',
```

Comment:
as discussed IRL you can remove it in the end, sorry

can you also add a comment saying that if `res_partner_id` si set, `unpartnered_email` will be ignore ? (and also double check it's the case + maybe a small test)

Reply:
I think I just should have kept the first half of the initial constraint.. I'll first check if doing that fixes it (probably will), if not I'll do that.